data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/CsaProfileCrac.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.data.crac_creation.creator.csa_profile;
+
+import com.farao_community.farao.data.crac_creation.creator.csa_profile.crac_creator.CsaProfileConstants;
+import com.farao_community.farao.data.native_crac_api.NativeCrac;
+import com.powsybl.triplestore.api.PropertyBags;
+import com.powsybl.triplestore.api.QueryCatalog;
+import com.powsybl.triplestore.api.TripleStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * @author Jean-Pierre Arnould {@literal <jean-pierre.arnould at rte-france.com>}
+ */
+public class CsaProfileCrac implements NativeCrac {
+
+    private final TripleStore tripleStoreCsaProfileCrac;
+
+    private final QueryCatalog queryCatalogCsaProfileCrac;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CsaProfileCrac.class);
+
+    public CsaProfileCrac(TripleStore tripleStoreCsaProfileCrac) {
+        this.tripleStoreCsaProfileCrac = tripleStoreCsaProfileCrac;
+        this.queryCatalogCsaProfileCrac = new QueryCatalog(CsaProfileConstants.SPARQL_FILE_CSA_PROFILE);
+    }
+
+    @Override
+    public String getFormat() {
+        return "CsaProfileCrac";
+    }
+
+    private PropertyBags queryTripleStore(String queryKey) {
+        String query = queryCatalogCsaProfileCrac.get(queryKey);
+        if (query == null) {
+            LOGGER.warn("Query [{}] not found in catalog", queryKey);
+            return new PropertyBags();
+        }
+        return tripleStoreCsaProfileCrac.query(query);
+    }
+
+    private PropertyBags queryTripleStore(List<String> queryKeys) {
+        PropertyBags mergedPropertyBags = new PropertyBags();
+        for (String queryKey : queryKeys) {
+            mergedPropertyBags.addAll(queryTripleStore(queryKey));
+        }
+        return mergedPropertyBags;
+    }
+
+    public PropertyBags getContingencies() {
+        return this.queryTripleStore(Arrays.asList(CsaProfileConstants.REQUEST_ORDINARY_CONTINGENCY, CsaProfileConstants.REQUEST_EXCEPTIONAL_CONTINGENCY, CsaProfileConstants.REQUEST_OUT_OF_RANGE_CONTINGENCY));
+    }
+
+    public PropertyBags getContingencyEquipments() {
+        return this.queryTripleStore(CsaProfileConstants.REQUEST_CONTINGENCY_EQUIPMENT);
```

Comment:
Filter et join tables in sql would be more efficient if I use only one request, but filtering in java is useful to log every error case.
And if I use a request by rdfId for example, it would be less efficient, too much sql calls if we have a large number of contingencies.